numpy/lib/tests/test_format.py

```diff
@@ -694,6 +694,29 @@ def test_write_version():
     '',
 ])
 
+def test_read_magic():
+    s1 = BytesIO()
+    s2 = BytesIO()
+
+    arr = np.ones((3, 6), dtype=float)
+
+    format.write_array(s1, arr, version=(1, 0))
+    format.write_array(s2, arr, version=(2, 0))
+
+    s1.seek(0)
+    s2.seek(0)
+
+    major1, minor1 = format.read_magic(s1)
+    major2, minor2 = format.read_magic(s2)
+
+    assert major1 == 1
```

Comment:
Don't use `assert`, it disappears when python is run with `-O`. Use `assert_` instead. You can also do this
```
version1 = format.read_magic(s1)
assert_(version1 == (1, 0))
```